frontends/web/src/utils/qttransport.ts

```diff
@@ -15,20 +16,21 @@
  */
 
 import { QWebChannel } from './qwebchannel';
+import { TMsgCallback, TQueryPromiseMap } from './transport-common';
 import { runningInQtWebEngine } from './env';
 
-let cache = null;
-let webChannel = null;
-let queryID = 0;
-let queryPromises = {};
-let currentListeners = [];
+let cache: any = null;
```

Comment:
after this PR is merged, I have some ideas how to improve this. `cache` can actually be `isChannelInitialized: boolean` instead, and `initTransport()` etc. could be made async functions that are much easier to read.